plugins/module_utils/redfish_utils.py

```diff
@@ -1834,12 +1834,16 @@ def get_fan_inventory(self):
                 result['ret'] = True
                 data = response['data']
 
-                for device in data[u'Fans']:
-                    fan = {}
-                    for property in properties:
-                        if property in device:
-                            fan[property] = device[property]
-                    fan_results.append(fan)
+                # Checking if fans are present
+                if 'Fans' in data:
```

Comment:
```suggestion
                if 'uFans' in data:
```
After all the iteration is done on `data[u'Fans']` and not on `data['Fans']`.